peps/pep-0791.rst

```diff
@@ -0,0 +1,158 @@
+PEP: 791
+Title: imath --- module for number-theoretic functions
+Author: Sergey B Kirpichev <skirpichev@gmail.com>
+Sponsor: Victor Stinner <vstinner@python.org>
+Discussions-To: Pending
+Status: Draft
+Type: Standards Track
+Created: 12-May-2025
+Python-Version: 3.15
+Post-History: `12-Jul-2018 <https://mail.python.org/archives/list/python-ideas@python.org/thread/YYJ5YJBJNCVXQWK5K3WSVNMPUSV56LOR/>`__,
+              02-Jun-2019,
+              `09-May-2025 <https://discuss.python.org/t/91337>`__,
+
+
+Abstract
+========
+
+This PEP proposes a new module for number-theoretical, combinatorial and other
+integer-valued functions defined for integer arguments, like
+:external+py3.14:func:`math.gcd` or :external+py3.14:func:`math.isqrt`.
+
+
+Motivation
+==========
+
+The :external+py3.14:mod:`math` documentation says: "This module provides access
+to the mathematical functions defined by the C standard."  But,
+over time the module was populated with functions that aren't related to
+the C standard or floating-point arithmetics.  Now it's much harder to describe
+module scope, content and interfaces (returned values or accepted arguments).
+
+For example, the :external+py3.14:mod:`math` module documentation says: "Except
+when explicitly noted otherwise, all return values are floats."  This is no
+longer true:  *None* of the functions listed in the `Number-theoretic
+functions <https://docs.python.org/3.14/library/math.html#number-theoretic-functions>`_
+subsection of the documentation returns a float, but the
+documentation doesn't say so.  In the proposed module a similar sentence "All
+return values are integers." could tell the truth once.  In a similar way we
+can simplify the description of the accepted arguments for functions in both the
+:external+py3.14:mod:`math` and the new module.
+
+Apparently, the :external+py3.14:mod:`math` can't serve as a catch-all place
+for mathematical functions: we have also the :external+py3.14:mod:`cmath` and
+the :external+py3.14:mod:`statistics`.  Let's do the same for integer-related
+functions.  It would provide shared context, which reduces verbosity in the
+documentation and conceptual load.  It also aids discoverability through
+grouping related functions and IDEs suggesting helpful completions.
```

Comment:
Tense here is inconsistent: conditional "would provide" vs present tense "aids discoverability". Probably best to use present tense throughout. 
```suggestion
functions.  It provides shared context, which reduces verbosity in the
documentation and conceptual load.  It also aids discoverability through
grouping related functions and makes IDE suggestions more helpful.
```